### ISSUE TITLE
WhatsProt->processInboundDataNode( ) 

### ISSUE THREAD

**Issue by hugodose** [mgp25/WhatsAPI-Official#563]
My Code:
```php
$debug = true;
$nickname = 'xxxx';                         // This is the username displayed by WhatsApp clients.
$username = 'xxxxxxxx';                     
$password = "xxxxxxxxxxxxxx"; 
$w = new WhatsProt($username, $nickname, $debug);
$w->connect(); // Connects to WhatsApp
$w->loginWithPassword($password); // Login
$w->sendGetPrivacyBlockedList(); // Get our privacy list
$w->sendGetServerProperties(); // Get server properties
$w->sendGetGroups(); // Get groups (participating)
$w->sendGetBroadcastLists(); // Get broadcasts lists
$w->sendPing(); // Keep alive
$w->PollMessage();
```
```
After getting the broadcast lists, I got this error:
Warning: Invalid argument supplied for foreach() in C:\wamp\www\WhatsAPI-Official-master\src\whatsprot.class.php on line 3174
```

**Comment by mgp25** [comment]
@hugodose full debug log!! post debug full log an full error stacktrace

**Comment by hugodose** [comment]
WooHoo!, Phone number XXXXXXXXXX connected successfully!
tx  <stream:features>
tx    <readreceipts></readreceipts>
tx    <groups_v2></groups_v2>
tx    <privacy></privacy>
tx    <presence></presence>
tx  </stream:features>

tx  <auth mechanism="WAUTH-2" user="556596282814">,+Üˆ556596282814JË€Æ£ Ì:ý¶Ë0¯9– y1427406014WhatsApp/2.12.68 S40Version/14.26 Device/Nokia302 MccMnc/724002</auth>

rx  <start from="s.whatsapp.net"></start>

rx  <stream:features></stream:features>

rx  <challenge>]YqžQ3Ú'À¤©ì™›dÃSø¿</challenge>

tx  <response>OÓŽÄNwšüÜüÈ={7¾ÌÏìuŸ6šXqð¡FÜ `_f:</response>

rx  <success t="1427406013" props="4" kind="free" status="active" creation="1427320854" expiration="1458856854">BÀdd`c½ò_ÁZb »ÚpF</success>

tx  <presence name="xxxx"></presence>

fim logintx  <iq id="getprivacy-1427406014-1" xmlns="jabber:iq:privacy" type="get">
tx    <query>
tx      <list name="default"></list>
tx    </query>
tx  </iq>

rx  <message from="xxxxxxxxxx@s.whatsapp.net" id="1427403499-323" offline="0" type="text" t="1427405990" notify="Piazzo">
rx    <body>kkk</body>
rx  </message>

Message from Piazzo: kkk tx  <receipt type="read" to="xxxxxxxxx@s.whatsapp.net" id="1427403499-323" t="1427406014"></receipt>

rx  <ib from="s.whatsapp.net">
rx    <offline count="6"></offline>
rx  </ib>

rx  <presence from="554884322670@s.whatsapp.net"></presence>

rx  <ack from="558498761645@s.whatsapp.net" id="1427405424-1" class="receipt" type="read"></ack>

rx  <iq from="556596282814@s.whatsapp.net" id="getprivacy-1427406014-1" type="error">
rx    <error code="404" text="item-not-found"></error>
rx  </iq>

tx  <iq id="getproperties-1427406015-2" type="get" xmlns="w" to="s.whatsapp.net">
tx    <props></props>
tx  </iq>

tx  <iq id="getgroups-1427406015-3" type="get" xmlns="w:g2" to="g.us">
tx    <participating></participating>
tx  </iq>

rx  <iq from="s.whatsapp.net" id="getproperties-1427406015-2" type="result">
rx    <props version="4">
rx      <prop name="timeout" value="300"></prop>
rx      <prop name="location" value="0"></prop>
rx      <prop name="readreceipts" value="1415214000"></prop>
rx      <prop name="qr" value="0"></prop>
rx      <prop name="groups_v2" value="1"></prop>
rx      <prop name="lists" value="1"></prop>
rx      <prop name="library" value="0"></prop>
rx      <prop name="audio" value="1"></prop>
rx      <prop name="checkmarks" value="0"></prop>
rx      <prop name="newmedia" value="0"></prop>
rx      <prop name="image_max_kbytes" value="5120"></prop>
rx      <prop name="image_quality" value="80"></prop>
rx      <prop name="image_max_edge" value="1280"></prop>
rx      <prop name="media" value="16"></prop>
rx      <prop name="broadcast" value="50"></prop>
rx      <prop name="max_subject" value="25"></prop>
rx      <prop name="max_participants" value="101"></prop>
rx      <prop name="max_groups" value="9999"></prop>
rx    </props>
rx  </iq>

rx  <iq from="g.us" id="getgroups-1427406015-3" type="result">
rx    <groups></groups>
rx  </iq>

tx  <iq id="get_lists_-1427406015-4" xmlns="w:b" type="get" to="s.whatsapp.net">
tx    <lists></lists>
tx  </iq>

tx  <iq id="ping_-1427406015-5" xmlns="w:p" type="get" to="s.whatsapp.net">
tx    <ping></ping>
tx  </iq>

rx  <iq from="s.whatsapp.net" id="get_lists_-1427406015-4" type="result">
rx    <lists>
rx      <list id="1427405698@broadcast" name="">
rx        <recipient jid="xxxxxxxx@s.whatsapp.net"></recipient>
rx        <recipient jid="xxxxxxxx@s.whatsapp.net"></recipient>
rx      </list>
rx    </lists>
rx  </iq>


( ! ) Warning: Invalid argument supplied for foreach() in C:\wamp\www\WhatsAPI-Official-master\src\whatsprot.class.php on line 3174
Call Stack
#	Time	Memory	Function	Location
1	0.0010	253456	{main}( )	..\exampleRegister.php:0
2	1.3981	2683384	WhatsProt->pollMessage( )	..\exampleRegister.php:64
3	1.5951	2688880	WhatsProt->processInboundData( )	..\whatsprot.class.php:586
4	1.8941	2953120	WhatsProt->processInboundDataNode( )	..\whatsprot.class.php:2598

**Comment by mgp25** [comment]
@hugodose try this:

```php
$debug = true;
$nickname = 'xxxx';                         // This is the username displayed by WhatsApp clients.
$username = 'xxxxxxxx';                     
$password = "xxxxxxxxxxxxxx"; 
$w = new WhatsProt($username, $nickname, $debug);
$w->connect(); // Connects to WhatsApp
$w->loginWithPassword($password); // Login
$w->pollMessage();
$w->sendGetPrivacyBlockedList(); // Get our privacy list
$w->pollMessage();
$w->sendGetServerProperties(); // Get server properties
$w->pollMessage();
$w->sendGetGroups(); // Get groups (participating)
$w->pollMessage();
$w->sendGetBroadcastLists(); // Get broadcasts lists
$w->pollMessage();
```

**Comment by hugodose** [comment]
Problem persists.

I commented out $w->sendGetBroadcastLists(); and now I don't see the error anymore.

I was trying do delete some BroadcastLists earlier, but didn't work.. so maybe I messed up with something. Maybe if I learn how to delete all my BroadcastLists it will solve the problem. 
It is really annoying to have a new broadcastlist ID for each message I send.... list is getting bigger and bigger...
